meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -30,19 +44,65 @@ interface IShowStyleVariantsProps {
 interface IShowStyleVariantsSettingsState {
 	editedMappings: ProtectedString<any>[]
 	timestampedFileKey: number
+	dndVariants: ShowStyleVariant[]
 }
 
+interface DragVariant {
+	index: number
+	type: ShowStyleDragDropTypes
+}
+
+const SET_STATE_DELAY: number = 50
+
 export const ShowStyleVariantsSettings = withTranslation()(
 	class ShowStyleVariantsSettings extends React.Component<
 		Translated<IShowStyleVariantsProps>,
 		IShowStyleVariantsSettingsState
 	> {
+		private timer: number | undefined
+
 		constructor(props: Translated<IShowStyleVariantsProps>) {
 			super(props)
 
 			this.state = {
 				editedMappings: [],
 				timestampedFileKey: Date.now(),
+				dndVariants: this.props.showStyleVariants,
+			}
+		}
+		componentDidUpdate(prevProps: Readonly<Translated<IShowStyleVariantsProps>>) {
+			if (this.showStyleVariantsChanged(prevProps) || this.noShowStyleVariantsPresentInState()) {
+				this.timer = Meteor.setTimeout(() => {
+					this.setState({
+						dndVariants: this.props.showStyleVariants,
+					})
+				}, SET_STATE_DELAY)
+			} else if (this.timer) {
+				Meteor.clearTimeout(this.timer)
+			}
+		}
+
+		componentWillUnmount() {
+			if (this.timer) {
+				Meteor.clearTimeout(this.timer)
+			}
+		}
+
+		private showStyleVariantsChanged = (prevProps: Readonly<Translated<IShowStyleVariantsProps>>): boolean => {
+			if (prevProps.showStyleVariants.length !== this.props.showStyleVariants.length) {
+				return true
+			}
+
+			if (this.props.showStyleVariants.length > 0 && this.state.dndVariants.length === 0) {
+				return true
+			}
+
+			return prevProps.showStyleVariants !== this.props.showStyleVariants && this.state.editedMappings.length > 0
+		}
+
+		private noShowStyleVariantsPresentInState = () => {
+			if (this.props.showStyleVariants.length > 0 && this.state.dndVariants.length === 0) {
+				return true
 			}
 		}
```

Comment:
I guess the return type of this should be boolean.
It can be reduced to:
```suggestion
		private noShowStyleVariantsPresentInState = () => {
			return this.props.showStyleVariants.length > 0 && this.state.dndVariants.length === 0
		}
```